app/custom_queries/hyrax/wings/custom_query_service.rb

```diff
@@ -0,0 +1,44 @@
+module Hyrax
+  module Wings
+    class CustomQueryService
+      # Defines custom queries for wings
+      # Use:
+      # => @custom_queries ||= Hyrax::Wings::CustomQueryService.new(query_service: ::Valkyrie.config.metadata_adapter.query_service)
+      # => resources = @custom_queries.find_many_by_alternate_ids(alternate_ids: ids, use_valkyrie: true)
+
+      def self.queries
+        [:find_many_by_alternate_ids]
+      end
+
+      attr_reader :query_service
+
+      def initialize(query_service:)
+        @query_service = query_service
+        @resource_factory = query_service.adapter.resource_factory
+      end
+
+      # implements a combination of two Valkyrie queries:
+      # => find_many_by_ids & find_by_alternate_identifier
+      # @param [Array<Valkyrie::ID, String>] ids
+      # @param [boolean] defaults to true; optionally return ActiveFedora::Base objects if false
+      # @return [Valkyrie::Resource or ActiveFedora::Base]
+      # NOTE: Ignores non-existent ids without raising an error
+      def find_many_by_alternate_ids(alternate_ids:, use_valkyrie: true)
+        alternate_ids.each do |id|
+          id = ::Valkyrie::ID.new(id.to_s) if id.is_a?(String)
+          raise ArgumentError, 'id must be a Valkyrie::ID' unless id.is_a? ::Valkyrie::ID
+        end
+        resources = []
+        alternate_ids.uniq.map(&:to_s).each do |id|
+          begin
+            af_object = ActiveFedora::Base.find(id)
```

Comment:
Could we reuse the existing bulk find, instead of querying each object individually?

Something like:

```ruby
af_objects = ActiveFedora::Base.find(alternate_ids.map(&:to_s))

return af_objects unless use_valkyrie

af_objects.map do |obj|
  @resource_factory.to_resource(object: af_object)
end
```